src/main/java/pl/rspective/voucherify/client/model/VoucherValidity.java

```diff
@@ -0,0 +1,137 @@
+package pl.rspective.voucherify.client.model;
+
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Class represents validity of a voucher
+ */
+public class VoucherValidity {
+
+  /**
+   * The unique voucher's code used for instance to consume it
+   */
+  private String code;
+
+  /**
+   * Status flag informing about validity of a voucher
+   */
+  private boolean valid;
+
+  /**
+   * Voucher's type
+   */
+  private VoucherType type;
+
+  /**
+   * Discount definition - present if type is DISCOUNT_VOUCHER
+   */
+  private Discount discount;
+
+  /**
+   * Gift definition (amount) - present if type is GIFT_VOUCHER
+   */
+  private Gift gift;
+
+  /**
+   * Tracking id which can be used in further calls
+   */
+  @SerializedName("tracking_id")
+  private String trackingId;
+
+  private VoucherValidity() {
+  }
+
+  private VoucherValidity(
+    String code,
+    boolean valid,
+    VoucherType type,
+    Discount discount,
+    Gift gift,
+    String trackingId) {
```

Comment:
When voucher is invalid then the response contains an additional field `reason`. I don't see it here.